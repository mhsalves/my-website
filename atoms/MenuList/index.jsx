import React from 'react';
import {
  arrayOf, bool, func, shape, string,
} from 'prop-types';

import { keyPrefixGenerator, keyMount } from '../../helpers/uniqueKey';

import {
  MenuListContainer,
  MenuListItem,
} from './styles';

const keyPrefix = keyPrefixGenerator();

function MenuList(props) {
  const { centered, items } = props;

  return (
    <MenuListContainer>
      {
        items.map(({ title, action }) => (
          <MenuListItem
            key={keyMount(keyPrefix, title)}
            onClick={action}
            centered={centered}
          >
            {title}
          </MenuListItem>
        ))
      }
    </MenuListContainer>
  );
}

MenuList.propTypes = {
  centered: bool,
  items: arrayOf(shape({
    title: string,
    action: func,
  })),
};

MenuList.defaultProps = {
  centered: false,
  items: [],
};

export default MenuList;
