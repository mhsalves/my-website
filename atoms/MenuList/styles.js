import styled, { css } from 'styled-components';

const MenuListContainer = styled.ul`
  width: 100%;
`;

const MenuListItem = styled.li`
  ${({ centered }) => centered && css`text-align: center;`}

  ${({ theme: { typography } }) => typography.regular}
  font-weight: 600;
  text-transform: uppercase;
  cursor: pointer;
  padding: ${({ theme: { space } }) => space.micro};
  color: ${({ theme: { colors } }) => colors.silver.dark};

  &:hover {
    color: ${({ theme: { colors } }) => colors.silver.default};
  }
`;

export {
  MenuListContainer,
  MenuListItem,
};
