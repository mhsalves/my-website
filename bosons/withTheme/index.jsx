import React from 'react';
import { ThemeProvider } from 'styled-components';
import baseTheme from '../themes/base';

const withTheme = Component => props => (
  <ThemeProvider theme={baseTheme}>
    <Component {...props} />
  </ThemeProvider>
);

export default withTheme;
