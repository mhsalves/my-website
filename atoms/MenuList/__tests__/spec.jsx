import React from 'react';
import renderer from 'react-test-renderer';

import MenuList from '../index';

import withTheme from '../../../bosons/withTheme';

const ComponentWithTheme = withTheme(MenuList);

const action = jest.fn();
const items = (new Array(5)).fill(null).map((v, i) => ({
  title: `Option ${i}`,
  action,
}));

describe('given: MenuList', () => {
  test('when: is center', () => {
    const component = renderer.create(
      <ComponentWithTheme items={items} centered />,
    ).toJSON();

    expect(component).toMatchSnapshot();
  });
});
