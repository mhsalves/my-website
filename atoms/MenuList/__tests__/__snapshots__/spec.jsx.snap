// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`given: MenuList when: is center 1`] = `
.c0 {
  width: 100%;
}

.c1 {
  text-align: center;
  font-size: 1rem;
  line-height: 1.5rem;
  font-weight: 400;
  margin: 0;
  padding: 0;
  font-family: 'Open Sans',serif;
  font-weight: 600;
  text-transform: uppercase;
  cursor: pointer;
  padding: 8px;
  color: #E8E6F0;
}

.c1:hover {
  color: #F2F0F7;
}

<ul
  className="c0"
>
  <li
    className="c1"
    onClick={[MockFunction]}
  >
    Option 0
  </li>
  <li
    className="c1"
    onClick={[MockFunction]}
  >
    Option 1
  </li>
  <li
    className="c1"
    onClick={[MockFunction]}
  >
    Option 2
  </li>
  <li
    className="c1"
    onClick={[MockFunction]}
  >
    Option 3
  </li>
  <li
    className="c1"
    onClick={[MockFunction]}
  >
    Option 4
  </li>
</ul>
`;
